app/models/miq_request_task/state_machine.rb

```diff
@@ -21,29 +21,15 @@ def signal(phase)
     begin
       send(phase)
     rescue => err
-      if [:finish, :provision_error].include?(phase.to_sym)
-        _log.error("[#{err.message}] encountered during [#{phase}]")
+      case phase
+      when :finish
+        _log.error("[#{err.class}: #{err.message}] encountered during [#{phase}]")
         _log.log_backtrace(err)
+      when :provision_error
+        update_and_notify_parent(:state => "finished", :status => "Error", :message => err.message)
+        signal(:finish)
       else
-        phase_context.delete(:error_message)
-
-        # HACK: Psych is unable to deserialize an Exception that contains a
-        #   NameError::message as its message.  When deserialized, it will raise
-        #   "TypeError: allocator undefined for NameError::message".  Calling
-        #   .message will dereference it and turn it into a String which is
-        #   serializable.  For more information about NameError::message, see
-        #   http://www.carboni.ca/blog/p/Ruby-Did-You-Know-That-5-NameErrormessage
-        #
-        #   In addition, backtrace information is not available, because it is
-        #   not an instance variable, but is instead dynamically generated.
-        #
-        #   As such, we store the exception's components separately in the
-        #   phase context.
-        phase_context[:exception_class]     = err.class.name
-        phase_context[:exception_message]   = err.message
-        phase_context[:exception_backtrace] = err.backtrace
-        phase_context[:error_phase]         = phase
-        signal :provision_error
+        signal(:provision_error)
```

Comment:
Where is the provision_error method for this to call, since it was removed below?

Reply:
Didn't see the point of it. Like you said, it unset everything in the `phase_context` that was set previously (which is why I removed that bit). I removed the logging because ultimately it would result in two log messages (one at `:provision_error`, one on `:finish`) which seemed redundant.

That just leaves calling `update_and_notify_parent` plus `signal(:finish)`. Rather than make a separate method, I just inlined it.